### PR TITLE
port more of the combinators

### PR DESCRIPTION
**Pull request by jrudolph**
A few of the important combinators `|`, `zeroOrMore`, `oneOrMore`, part of `~>`, `push`, and `drop`

### CODE REVIEW

**Comment by jrudolph** [comment]
Next bit I'm working on is calling other rules, after which it might be possible to run `CalculatorSpec`.

**Comment by jrudolph** [comment]
Now `CalculatorSpec` mostly works and only fails a error reporting test!

**Comment by jrudolph** [comment]
All combinators done! (I didn't implement all the optimizations for `times` yet, but that should be easy to add later on)

File: parboiled-core/src/main/scala-3/org/parboiled2/ParserMacros.scala [hunk 1]
```diff
@@ -295,8 +718,36 @@ class OpTreeContext(parser: Expr[Parser])(using Quotes) {
...
+    def collector(lifter: Term): Collector =
+      lifter match {
+        case TypeApply(Ident("forRule0" | "forReduction"), _) => rule0Collector
+        case TypeApply(Ident("forRule1"), _)                  => rule1Collector
+        case x                                                => reportError(s"Unexpected lifter ${lifter.show(using Printer.TreeStructure)}", lifter.asExpr)
+      }
+
+    // a list of names for operations that are not yet implemented but that should not be interpreted as rule calls
+    // FIXME: can be removed when everything is implemented
+    val ruleNameBlacklist =
```
Comment (yanns) [inline parboiled-core/src/main/scala-3/org/parboiled2/ParserMacros.scala:735]: 👍  good way to make progress.

**Review by yanns — APPROVED** [review]
💯

**Pull request merged**